cloudoperators/juno
# fix(ui): fixes z index stacking problems for PageHeader

# Summary

This PR fixes the z-index stacking problem introduced by the recent changes to `PortalProvider`. The main change introduced with the new `PortalProvider` was that every portaled component now renders in its own container with `z-index: 1` within the portal root element. The thought is to ensure a level playing field for all portaled components, especially when multiple portaled components are open at the same time to ensure that they stack properly. 

This has the side effect that components which aren't rendered in a portal will render above all portaled components if they have a z-index higher than 1. This happened with the `PageHeader` component. It had a `z-index` of 50 which means it was rendered above the `Panel`'s header, obscuring it.

# Changes Made

- Changed `PageHeader` `z-index` to 1 to ensure proper stacking with `Panel` and `Modal`
- Removed `z-index` from `Label`. It had a `z-index` of 10 which meant it would have been above `PageHeader` when scrolling down. We might have to revisit this again if it turns out that this breaks other edge cases with the `Label`
- Added `PortalProvider` decorator to the `Panel`, `PanelBody`, and `PanelFooter` stories to ensure the Portal renders in a portal root and gets the correct styling

# Related Issues

- https://github.com/cloudoperators/juno/pull/513
- https://github.com/cloudoperators/juno/pull/540


# Testing Instructions

1. cd apps/example
2. `npx turbo dev`
3. click on edit item to open panel
4. PanelHeader should be visible and not obscured by PageHeader

To test Panel stories check storybook

# Checklist
